momepy/preprocessing.py

```diff
@@ -1105,3 +1109,447 @@ def roundabout_simplification(
     output = _ext_lines_to_center(edges, incoming_all, idx_drop)
 
     return output
+
+
+def consolidate_intersections(
+    graph,
+    tolerance=30,
+    rebuild_graph=True,
+    rebuild_edges_method="spider",
+    x_att="x",
+    y_att="y",
+    edge_from_att="from",
+    edge_to_att="to",
+):
+    """
+    Consolidate close street intersections into a single node, collapsing short edges.
+
+    If rebuild_graph is True, new edges are drawn according to ``rebuild_edges_method`` which is one of:
+
+    1. Extension reconstruction:
+        Edges are linearly extended from original endpoints until the new nodes. This method preserves
+        most faithfully the network geometry but can result in overlapping geometry.
+    2. Spider-web reconstruction:
+        Edges are cropped within a buffer of the new endpoints and linearly extended from there. This
+        method improves upon linear reconstruction by mantaining, when possible, network planarity.
+    3. Euclidean reconstruction:
+        Edges are ignored and new edges are built as straight lines between new origin and new
+        destination. This method ignores geometry, but efficiently preserves adjacency.
+
+    If ``rebuild_graph`` is False, graph is returned with consolidated nodes but without reconstructed
+    edges i.e. graph is intentionally disconnected.
+
+    Graph must be configured so that
+
+    1. All nodes have attributes determining their x and y coordinates;
+    2. All edges have attributes determining their origin, destination, and geometry.
+
+    Parameters
+    ----------
+    graph : Networkx.Graph, Networkx.DiGraph, Networkx.MultiGraph, or Networkx.MultiDiGraph
+    tolerance : float, default 30
+        distance in network units below which nodes will be consolidated
+    rebuild_graph: Boolean
+    rebuild_edges_method: string
+        'extend' or 'spider' or 'euclidean', ignored if rebuild_graph is False
+    x_att: string
+        node attribute with the valid x-coordinate
+    y_att: string
+        node attribute with the valid y-coordinate
+    edge_from_att: string
+        edge attribute with the valid origin node id
+    edge_to_att: string
+        edge attribute with the valid destination node id
+
+    Returns
+    ----------
+    Networkx.MultiGraph or Networkx.MultiDiGraph
+        directionality inferred from input type
+
+    """
+    # Collect nodes and their data:
+    nodes, nodes_dict = zip(*graph.nodes(data=True))
+    nodes_df = pd.DataFrame(nodes_dict, index=nodes)
+    nodes_geometries = gpd.points_from_xy(nodes_df[x_att], nodes_df[y_att])
+    graph_crs = graph.graph.get("crs")
+    nodes_gdf = gpd.GeoDataFrame(
```

Comment:
From [`linting`](https://github.com/pysal/momepy/actions/runs/3237046086/jobs/5303895833#step:4:65) we can see that `nodes_gdf` is declared, but not actually used anywhere.